CRM/PivotReport/Entity.php

```diff
@@ -16,7 +18,25 @@ class CRM_PivotReport_Entity {
       ),
     ),
     'Membership' => array(),
-    'Contact' => array(),
+    'People' => array(
+      'extensions' => array(
+        'org.civicrm.hrjobcontract'
+      ),
+    ),
+    'Leave' => array(
+      'extensions' => array(
+        'uk.co.compucorp.civicrm.hrleaveandabsences',
+        'org.civicrm.hrjobcontract'
+      ),
+    ),
+//    'Sample' => array(
+//      'hookable' => [
+//        'namespace' => 'HRLeaveAndAbsences',
+//        'extension' => 'uk.co.compucorp.civicrm.hrleaveandabsences',
+//        'template_path' => 'templates/CRM/HRLeaveAndAbsences/Page/SampleReport.tpl'
+//      ]
```

Comment:
I think this can be simplified a bit. Instead of passing a namespace, I think we can just have something like:

```php 
  'data_class' => 'CRM_MyExtension_SomeFolder_DataClass',
  'group_class' => 'CRM_MyExtension_SomeFolder_GroupClass'
```

This will give people even more flexibility and allow them to name the classes whichever way they want.

The same applies to the template. You can just pass `CRM/HRLeaveAndAbsences/Page/SampleReport.tpl` and should be possible to use it. You don't really a full path, which means you probably don't need the `extension` too in order to figure out the full path

Reply:
This is a great Idea and makes the whole class naming process flexible. I tested it and it works fine.